ttnn/cpp/ttnn/deprecated/tt_dnn/op_library/nlp_tms/nlp_tms.hpp

```diff
@@ -93,22 +93,6 @@ struct NlpCreateHeadsDecode {
         const std::vector<Tensor>& input_tensors, std::vector<Tensor>& output_tensors) const;
 };
 
-struct NlpCreateHeads {
-    const uint32_t num_q_heads;
-    const uint32_t num_kv_heads;
-    const uint32_t head_dim;
-    const bool transpose_k_heads;
-    MemoryConfig output_mem_config;
-
-    void validate(const std::vector<Tensor>& input_tensors, const std::vector<std::optional<const Tensor>>& optional_input_tensors) const;
-    std::vector<Shape> compute_output_shapes(const std::vector<Tensor>& input_tensors) const;
-    std::vector<Tensor> create_output_tensors(const std::vector<Tensor>& input_tensors) const;
-    operation::ProgramWithCallbacks create_program(
-        const std::vector<Tensor>& input_tensors,
-        const std::vector<std::optional<const Tensor>>& optional_input_tensors,
-        std::vector<Tensor>& output_tensors) const;
-};
-
 struct NlpConcatHeads {
```

Comment:
the rest should be moved to to close this ticket, right?

Reply:
yeah this ticket encapsulates all of them...it has taken me some time  to do just this one with the new infra (required some changes to support multiple tensors) so have not done it all in one go. The next few will be quicker